migrate download from POST to GET

it is easy to hit error `http: server closed idle connection` on downloading. `POST` means non retry-able in go and http standard . https://stackoverflow.com/questions/42847294/how-to-catch-http-server-closed-idle-connection-error


gosdk/zboxcli should be updated too.

```
0box-sdk        [INFO]   2021/12/17 09:16:10.165982 downloadworker.go:254: Encrypted key from fileref
 0 / 31101916 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]   0.00%0box-sdk        [INFO]   2021/12/17 09:16:10.166295 downloadworker.go:285: Download Size:31101916 Shard:4443131 chunks/shard:68
0box-sdk        [INFO]   2021/12/17 09:16:10.166318 downloadworker.go:286: Start block: 1 End block: 68 Num blocks: 10
0box-sdk        [INFO]   2021/12/17 09:16:10.166618 downloadworker.go:298: Downloading block 1
0box-sdk        [INFO]   2021/12/17 09:16:10.166689 downloadworker.go:104: downloadBlock 1 numDownloads 9
 0 / 31101916 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]   0.00%0box-sdk        [INFO]   2021/12/17 09:16:11.048538 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [INFO]   2021/12/17 09:16:11.097151 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [INFO]   2021/12/17 09:16:11.101627 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [INFO]   2021/12/17 09:16:11.102198 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [ERROR]  2021/12/17 09:16:11.104062 downloadworker.go:118: Download block : http://huawei.0chain.net:31306 Post "http://huawei.0chain.net:31306/v1/file/download/5e63651cfbbe960d8e0d1fe81c9ab10a6b8012c806dd74ac564a659deea7d984": EOF
0box-sdk        [INFO]   2021/12/17 09:16:11.141352 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [INFO]   2021/12/17 09:16:11.145082 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [ERROR]  2021/12/17 09:16:11.147243 downloadworker.go:118: Download block : http://huawei.0chain.net:31304 Post "http://huawei.0chain.net:31304/v1/file/download/5e63651cfbbe960d8e0d1fe81c9ab10a6b8012c806dd74ac564a659deea7d984": http: server closed idle connection
0box-sdk        [INFO]   2021/12/17 09:16:11.147265 blockdownloadworker.go:227: Will be retrying download
0box-sdk        [INFO]   2021/12/17 09:16:11.158659 blockdownloadworker.go:227: Will be retrying download
 0 / 31101916 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]   0.00%0box-sdk        [INFO]   2021/12/17 09:16:11.190942 blockdownloadworker.go:227: Will be retrying download
 0 / 31101916 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]   0.00%0box-sdk        [INFO]   2021/12/17 09:16:29.688270 downloadworker.go:298: Downloading block 11
0box-sdk        [INFO]   2021/12/17 09:16:29.688370 downloadworker.go:104: downloadBlock 11 numDownloads 9
 4587520 / 31101916 [============================>-----------------------------------------------------------------------------------------------------------------------------------------------------------------------]  14.75% 01m45s0box-sdk        [INFO]   2021/12/17 09:16:37.615804 downloadworker.go:298: Downloading block 21
0box-sdk        [INFO]   2021/12/17 09:16:37.615883 downloadworker.go:104: downloadBlock 21 numDownloads 9
 9175040 / 31101916 [=========================================================>------------------------------------------------------------------------------------------------------------------------------------------]  29.50% 00m44s0box-sdk        [INFO]   2021/12/17 09:16:59.079934 downloadworker.go:298: Downloading block 31
0box-sdk        [INFO]   2021/12/17 09:16:59.080920 downloadworker.go:104: downloadBlock 31 numDownloads 9
 13762560 / 31101916 [======================================================================================>------------------------------------------------------------------------------------------------------------]  44.25% 00m44s0box-sdk        [INFO]   2021/12/17 09:17:17.153243 downloadworker.go:298: Downloading block 41
0box-sdk        [INFO]   2021/12/17 09:17:17.153338 downloadworker.go:104: downloadBlock 41 numDownloads 9
 18350080 / 31101916 [===================================================================================================================>-------------------------------------------------------------------------------]  59.00% 00m26s0box-sdk        [INFO]   2021/12/17 09:17:37.810521 downloadworker.go:298: Downloading block 51
0box-sdk        [INFO]   2021/12/17 09:17:37.811347 downloadworker.go:104: downloadBlock 51 numDownloads 9
 22937600 / 31101916 [===============================================================================================================================================>---------------------------------------------------]  73.75% 00m22s0box-sdk        [INFO]   2021/12/17 09:17:46.597212 downloadworker.go:298: Downloading block 61
0box-sdk        [INFO]   2021/12/17 09:17:46.597279 downloadworker.go:104: downloadBlock 61 numDownloads 9
 31101916 / 31101916 [====================================================================================================================================================================================================] 100.00% 1m54s
Status completed callback. Type = video/mp4. Name = BigBuckBunny4.mp4
```

--- a/code/go/0chain.net/blobbercore/handler/handler.go
+++ b/code/go/0chain.net/blobbercore/handler/handler.go
@@ -44,7 +44,7 @@ func SetupHandlers(r *mux.Router) {
 
 	//object operations
 	r.HandleFunc("/v1/file/upload/{allocation}", common.ToJSONResponse(WithConnection(UploadHandler)))
-	r.HandleFunc("/v1/file/download/{allocation}", common.ToByteStream(WithConnection(DownloadHandler))).Methods(http.MethodPost, http.MethodOptions)
+	r.HandleFunc("/v1/file/download/{allocation}", common.ToByteStream(WithConnection(DownloadHandler))).Methods(http.MethodGet, http.MethodOptions)
 	r.HandleFunc("/v1/file/rename/{allocation}", common.ToJSONResponse(WithConnection(RenameHandler))).Methods(http.MethodPost, http.MethodOptions)
 	r.HandleFunc("/v1/file/copy/{allocation}", common.ToJSONResponse(WithConnection(CopyHandler)))
 	r.HandleFunc("/v1/file/attributes/{allocation}", common.ToJSONResponse(WithConnection(UpdateAttributesHandler)))

--- a/code/go/0chain.net/blobbercore/handler/object_operation_handler_test.go
+++ b/code/go/0chain.net/blobbercore/handler/object_operation_handler_test.go
@@ -435,7 +435,7 @@ func TestDownloadFile(t *testing.T) {
 		body := new(bytes.Buffer)
 		formWriter := multipart.NewWriter(body)
 		rm := addToForm(t, formWriter, p)
-		req := httptest.NewRequest(http.MethodPost, "/v1/file/download/", body)
+		req := httptest.NewRequest(http.MethodGet, "/v1/file/download/", body)
 		require.NoError(t, req.ParseForm())
 		req.Header.Add("Content-Type", formWriter.FormDataContentType())
 		return req, rm

